pkg/datex/date.go

```diff
@@ -0,0 +1,64 @@
+// Package datex is a wrapper around github.com/rickb777/date/v2 and github.com/rickb777/period
+// so we don't depend on it directly.
+package datex
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rickb777/period"
+	"github.com/samber/lo"
+)
+
+type ISOString period.ISOString
+
+func (i ISOString) Parse() (Period, error) {
+	res, err := period.Parse(string(i))
+	return Period{res}, err
+}
+
+// ParsePtrOrNil parses the ISO8601 string representation of the period or if ISOString is nil, returns nil
+func (i *ISOString) ParsePtrOrNil() (*Period, error) {
+	if i == nil {
+		return nil, nil
+	}
+
+	d, err := i.Parse()
+	if err != nil {
+		return nil, err
+	}
+
+	return lo.ToPtr(d), nil
+}
+
+type Period struct {
+	period.Period
+}
+
+// FromDuration creates an IMPRECISE Period from a time.Duration
```

Comment:
why is it IMPRECISE?

Reply:
Right, sorry, with the Normalize(false) this will not be imprecise.